src/pykeen/training/training_loop.py

```diff
@@ -1028,6 +1046,9 @@ def _sub_batch_size_search(
             )
         except RuntimeError as runtime_error:
             self._free_graph_and_cache()
+            # TODO: hotfix for is_oom_error not properly handling UnsupportedWeight
+            if isinstance(runtime_error, NoSampleWeightSupportError | UnsupportedLabelSmoothingError):
+                raise
```

Comment:
This looks like something to fix upstream in `torch_max_mem`

Reply:
does this make sense? torch_max_mem shouldn't have to know about the errors we're only using in pykeen

Reply:
`torch_max_mem` seems to (incorrectly) assume that `error.args[0]` is of type `str` (`NoSampleWeightSupportError` is just one example which does not use a `str` there).

Reply:
https://github.com/mberr/torch-max-mem/pull/25 - however, I am now stuck with making an new release, as the old way seems to be outdated 😅 

-> https://github.com/mberr/torch-max-mem/pull/26

Reply:
included in https://github.com/pykeen/pykeen/pull/1534/commits/0910db6dc937e4ece92b2abf45a218f3a1a45cb8